internal/service/advancedcluster/validate_schema.go

```diff
@@ -105,3 +105,34 @@ func (v RegionSpecPriorityOrderDecreasingValidator) ValidateList(ctx context.Con
 		}
 	}
 }
+
+// UseEffectiveFieldsValidator validates that use_effective_fields is not set for Flex or Tenant clusters
+type UseEffectiveFieldsValidator struct{}
+
+func (v UseEffectiveFieldsValidator) Description(ctx context.Context) string {
+	return v.MarkdownDescription(ctx)
+}
+
+func (v UseEffectiveFieldsValidator) MarkdownDescription(_ context.Context) string {
+	return "use_effective_fields cannot be set for Flex or Tenant clusters"
```

Comment:
We have a mix of period usage in these MarkdownDescriptions. Just calling attention to it.
```suggestion
	return "use_effective_fields cannot be set for Flex or Tenant clusters."
```